Add support for distributed implementation in ripple reader (#376)
hyperspy/rosettasciio: doc/user_guide/lazy.rst

.. _lazy:

============
Lazy loading
============

.. _memory mapping: https://docs.dask.org/en/stable/array-creation.html#memory-mapping

> .. _dask distributer: https://distributed.dask.org

Author: ```suggestion
.. _dask distributed: https://distributed.dask.org
```

Data can be loaded lazily by using ``lazy=True``, however not all formats are :ref:`supported <supported-formats>`.
The data will be loaded as a dask array instead of a numpy array.

.. code:: python

    >>> from rsciio import msa
    >>> d = msa.file_reader("file.mrc", lazy=True)
    >>> d["data"]
    dask.array<array, shape=(10, 20, 30), dtype=int64, chunksize=(10, 20, 30), chunktype=numpy.ndarray>

Memory mapping
==============

Binary file formats are loaded lazily using `memory mapping`_.
The common implementation consists in passing the :class:`numpy.memmap` to the :func:`dask.array.from_array`.

> However, it has some shortcomings, to name a few: it is not compatible with the `dask distributer`_
> scheduler and it has limited control on the memory usage.

Reviewer: ```suggestion
However, it has some shortcomings, to name a few: it is not compatible with the `dask distributed`_
scheduler and it has limited control on the memory usage.
```

> For supported file format, a different implementation can be used to load data lazily in a manner that is

Reviewer: ```suggestion
For supported file formats, a different implementation can be used to load data lazily in a manner that is
```

> compatible with the `dask distributer`_  scheduler and allow better control of the memory usage. 

Reviewer: ```suggestion
compatible with the `dask distributed`_  scheduler and allow for better control of the memory usage. 
```

> This implementation uses an approach similar as describe in the dask documentation on

Reviewer: ```suggestion
This implementation uses an approach similar to that described in the dask documentation on
```

> `memory mapping`_ and be enabled using the ``distributed`` parameter (not all formats are

Reviewer: ```suggestion
`memory mapping`_ and is enabled using the ``distributed`` parameter (not all formats are
```
